### DISCUSSION TITLE
Suppressing unused variable diagnostic

### DISCUSSION THREAD

**Discussion by stephen-huan** [microsoft/pyright#7410] [Ideas]
```python
def f(
    x,  # pyright: ignore
):
    ...
```

no longer suppresses the diagnostic hint (from the LSP) that "x" is not accessed which I realized by bisecting to 6426ebccc0d83a3325d67096bdb00667c02b2d04. 

Instead, one can use

```python
def f(_):
    ...
```

which works for one argument but not multiple as it's impossible to declare a function with the header `f(_, _)`. 

Instead, one has the use the awkward pattern

```python
def f(x, y):
    (x, y)  # pyright: ignore [reportUnusedExpression]
    ...
```

Would it be possible to suppress the diagnostic hint for variables whose name starts with an underscore, like `reportUnusedVariable`? That would permit writing the natural

```python
def f(_x, _y):
    ...
```

>  **reportUnusedVariable** [boolean or string, optional]: Generate or suppress diagnostics for a variable that is not accessed. The default value for this setting is `"none"`. Variables whose names begin with an underscore are exempt from this check.

Alternatively, ignoring `"_"*n` also makes sense.
```python
def f(_, __):
    ...
```

**Comment by erictraut** [comment]
Pyright is working as designed here. Please read https://github.com/microsoft/pyright/issues/1118 if you are interested in details.

**Reply by stephen-huan** [reply]
I see, sorry for the noise!
